jofing-forDevReview/UserDataBot/UserDataBot.cs

```diff
@@ -0,0 +1,110 @@
+﻿namespace UserDataBot
+{
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Bot.Builder;
+    using Microsoft.Bot.Builder.Dialogs;
+    using Microsoft.Bot.Builder.Integration;
+    using Microsoft.Bot.Schema;
+    using Microsoft.Extensions.Options;
+
+    /// <summary>Defines the bot for the persisting user data tutorial.</summary>
+    public class UserDataBot : IBot
+    {
+        /// <summary>The state property accessor for user data.</summary>
+        private IStatePropertyAccessor<UserData> UserDataAccessor { get; }
+
+        /// <summary>The dialog set that has the dialog to use.</summary>
+        private GreetingsDialog GreetingsDialog { get; }
+
+        /// <summary>Create a new instance of the bot.</summary>
+        /// <param name="options">The options to use for our app.</param>
+        /// <param name="greetingsDialog">An instance of the dialog set.</param>
+        public UserDataBot(IOptions<BotFrameworkOptions> options, GreetingsDialog greetingsDialog)
```

Comment:
>IOptions<BotFrameworkOptions> options, GreetingsDialog greetingsDialog) [](start = 27, length = 71)

this should be simply
UserDataBot(UserState, ConversationState)

Then create your property using that.